File: data/EXP01/golang_no_system_var/2370/2370.output_1.go
func longestIdealString(s string, k int) int {
    n := len(s)
    dp := make([][]int, 26)
    for i := range dp {
        dp[i] = make([]int, k+1)
    }
    
    maxLen := 0
    for i := 0; i < n; i++ {
        for j := 0; j < 26; j++ {
            for l := 0; l <= k; l++ {
                if int(s[i]-'a') == j {
                    dp[j][l] = max(dp[j][l], l+1)
                } else {
                    dp[j][l] = max(dp[j][l], dp[int(s[i]-'a')][l]+1)
                }
            }
            maxLen = max(maxLen, dp[j][k])
        }
    }
    
    return maxLen
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: data/EXP01/python/2370/2370.output_1.py
# @lc app=leetcode id=2370 lang=python3
class Solution:
    def longestIdealString(self, s: str, k: int) -> int:
        n = len(s)
        dp = [1] * n
        
        for i in range(1, n):
            for j in range(i-1, -1, -1):
                diff = abs(ord(s[i]) - ord(s[j]))
                if diff <= k:
                    dp[i] = max(dp[i], dp[j] + 1)
        
        return max(dp)